packages/contracts/contracts/deploy/PayableDAppFactory.sol

```diff
@@ -0,0 +1,132 @@
+// SPDX-License-Identifier: Apache-2.0
+/// @title ERC20 DApp Factory
+pragma solidity ^0.8.13;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import {IConsensus} from "@cartesi/rollups/contracts/consensus/IConsensus.sol";
+import {CartesiDApp} from "@cartesi/rollups/contracts/dapp/CartesiDApp.sol";
+import {ICartesiDApp} from "@cartesi/rollups/contracts/dapp/ICartesiDApp.sol";
+
+import {ICartesiIPFSDAppFactory} from "../factory/ICartesiIPFSDAppFactory.sol";
+import {IFinancialProtocol} from "../protocol/IFinancialProtocol.sol";
+import {IMachineProtocol} from "../protocol/IMachineProtocol.sol";
+import {IPayableDAppFactory} from "./IPayableDAppFactory.sol";
+
+/// @notice This contract implements an IPayableDAppFactory based on a IERC20 token
+contract PayableDAppFactory is
+    IPayableDAppFactory,
+    IFinancialProtocol,
+    IMachineProtocol,
+    Ownable
+{
+    ICartesiIPFSDAppFactory public immutable factory;
+    IERC20 public immutable token;
+    IConsensus public immutable consensus;
+    uint256 public immutable price;
+
+    mapping(ICartesiDApp => uint256) public runway;
+
+    /// @notice Construct a new ERC20DAppFactory with the given token as payment method
+    /// @param _factory The factory to delegate calls to instantiate the dapp
+    /// @param _consensus The consensus contract to use
+    /// @param _price The price per second of execution of the dapp node
+    constructor(
+        address _owner,
+        ICartesiIPFSDAppFactory _factory,
+        IERC20 _token,
+        IConsensus _consensus,
+        uint256 _price
+    ) {
+        factory = _factory;
+        token = _token;
+        consensus = _consensus;
+        price = _price;
+
+        // transfer ownership to the give owner
+        transferOwnership(_owner);
+    }
+
+    /// @dev This function deploys a new application, and registers it with the payment system
+    function newApplication(
+        address _dappOwner,
+        bytes32 _templateHash,
+        string memory _cid,
+        uint256 _runway
+    ) external override returns (CartesiDApp) {
+        // delegate call to factory, owner must be this contract, so we can register with payment system
```

Comment:
Isn't it cheaper to create the DApp with the correct owner and use another method to register it with the payment system? Something like an event.

Might be worth testing :)

Reply:
Are you talking about the following `transferOwnership`?
I think we can get rid of the `transferOwnership` and use `_dappOwner` right away, right?

Reply:
Yeah! Sorry, marked the wrong line